src/sickle.ts

```diff
@@ -66,6 +77,34 @@ class Annotator {
           this.writeNode(node);
         }
         break;
+      case ts.SyntaxKind.InterfaceDeclaration:
+        this.writeRange(node.getFullStart(), node.getEnd());
+        if (node.flags & ts.NodeFlags.Ambient) {
+          // It's a "declare interface ..."; write a Closure interface
+          // declaration so we can @export all properties.
+          let decl = <ts.InterfaceDeclaration>node;
+          let className = decl.name.getText();
+          let props: TypedProperty[] = [];
+          for (let member of decl.members) {
+            // Members can include things like index signatures, for e.g.
+            //   interface Foo { [key: string]: number; }
+            // For now, just die unless all the members are regular old
+            // properties.
+            if (member.kind != ts.SyntaxKind.PropertySignature) {
+              this.fail(`unhandled member of kind ${ts.SyntaxKind[member.kind]}`);
+            }
+            let prop = <ts.PropertySignature>member;
+            props.push({
+              name: prop.name.getText(),
```

Comment:
The prop is actually different in each: one is a PropertySignature, one is a ParameterDeclaration, and one is a PropertyDeclaration.  They don't have a shared supertype and they don't have a shared type for their "name" field (e.g. you can have class properties with literal expressions for their names, whereas constructors can have a BindingPattern for their name field).

Re a class in general, I prefer the record because it gives named arguments (three arguments is dangerously close to being hard to track the proper arg order).